### PR TITLE
week2_priyasa

### PR DESCRIPTION
**Pull request by priyasa399**
First program (prime numbers)
Screenshot from 2022-12-28 15-48-23

Second program (Pet)
Screenshot from 2022-12-28 15-47-59

### CODE REVIEW

File: week_2/pet.rb [hunk 1]
```diff
@@ -110,6 +118,15 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    ans = {}
+    for pet in pets
```
Comment (hrushikeshj) [inline week_2/pet.rb:122]: `each` is preferred way of [iterating](https://dev.to/jolouie7/ruby-each-map-reduce-unless-and-until-3b5c) in Ruby.

File: week_2/pet.rb [hunk 2]
```diff
@@ -110,6 +118,15 @@ def self.cost_to_feed(pets, days)
...
+      if ans.keys.include? pet.animal_type_id
```
Comment (hrushikeshj) [inline week_2/pet.rb:123]: You can also just check if `ans[pet.animal_type_id]` is not nil

File: week_2/pet.rb [hunk 3]
```diff
@@ -90,7 +94,11 @@ def food_cost(days)
   # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
   # Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
   def self.cost_to_feed(pets, days)
-    raise NotImplementedError # TODO
+    sum = 0
+    for pet in pets
+      sum = sum + pet.food_consumed_per_day*days*FOOD_COST_PER_KG[pet.animal_type_id]
+    end
```
Comment (hrushikeshj) [inline week_2/pet.rb:100]: You can make use of the [`reduce`](https://medium.com/@james.a.hughes/using-the-reduce-method-in-ruby-907f3c18ae1f) method here.

**Review by hrushikeshj — APPROVED** [review]
Good Job

File: week_2/pet.rb [hunk 4]
```diff
@@ -46,7 +46,11 @@ def initialize(name: , animal_type_id: , food_consumed_per_day:)
 
   # Return the habitat of the pet
   def habitat
-    raise NotImplementedError # TODO
+    for key in HABITATS.keys
+      if HABITATS[key].include? animal_type_id
+        return key
+      end
+    end
```
Comment (hrushikeshj) [inline week_2/pet.rb:53]: You can make use of [`find`](https://apidock.com/ruby/Enumerable/find) method here.

File: week_2/pet.rb (hunk 2 above)
Comment (priyasa399) [reply week_2/pet.rb]: I have made the changes

File: week_2/pet.rb (hunk 3 above)
Comment (priyasa399) [reply week_2/pet.rb]: I have made the changes

File: week_2/pet.rb (hunk 4 above)
Comment (priyasa399) [reply week_2/pet.rb]: I have made the changes

File: week_2/pet.rb [hunk 5]
```diff
@@ -90,7 +90,7 @@ def food_cost(days)
   # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
   # Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
   def self.cost_to_feed(pets, days)
-    raise NotImplementedError # TODO
+    pets.reduce(0){|sum,pet| sum+pet.food_consumed_per_day*days*FOOD_COST_PER_KG[pet.animal_type_id]}
```
Comment (hrushikeshj) [inline week_2/pet.rb:93]: Keep your code [`DRY`](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

```suggestion
    pets.reduce(0) { |sum,pet| sum + pet.food_cost(days) }
```

Comment (priyasa399) [reply week_2/pet.rb]: Done

**Pull request closed without merging**